### ISSUE TITLE
Not possible anymore to index on spoint with NULL

### ISSUE THREAD

**Issue by gmantele** [akorotkov/pgsphere#9]
I used the ZIP file generated from the last commit (1a84decf3f734cdbd6db5eff95baec06152b802d, labeled "Compatibility with PostgreSQL 11") with a PostgreSQL-10 database. I have created a table with the [Hipparcos main table](http://vizier.u-strasbg.fr/viz-bin/VizieR-3?-source=I/239/hip_main&-out.max=50&-out.form=HTML%20Table&-out.add=_r&-out.add=_RAJ,_DEJ&-sort=_r&-oc.form=sexa) containing few rows with `ra` and `dec` set to `NULL`.

With a previous version of PgSphere, creating an index on `spoint(radians(ra), radians(dec))` worked very well with the following SQL statement:

```sql
CREATE INDEX hip_icrs_idx ON hipparcos                                           
       USING GIST (spoint(radians(ra), radians(dec)));
```

But with this last commit, it always fails.

I discovered that it was due to `NULL` values. `spoint(radians(NULL), radians(NULL)))` works well though ; it returns `NULL`. So, it seems the issue lies in the function used by the GiST index....probably some functions updated in the last commit, but I could not tell which one (I am no expect in GiST).

Maybe it is normal...after all, I used a version targeting PostgreSQL-11, not -10. But I expected it would be backward compatible.

So, for the moment I continue to use PgSphere with the previous commit (ac4527125cb6d3d1b479ff7d34c63104169c6af8).

**Comment by mnullmei** [comment]
Hi Grégory, thanks for the report!

OK, the problem should be solved with my "ad-hoc" commit 1fd7db81049cf0f26abe70acdf6d59bf7e47e8e3.

Did you successfully run "make installcheck" for pgSphere on your installation? For, on the configurations I tested, 1a84dec consistently failed for that. However, the table to be indexed apparently requires a certain size to fail. If "make installcheck" would pass somewhere for 1a84dec, the index-creation tests should be spruced up.

**Comment by gmantele** [comment]
I have just checked with your commit (1fd7db8): it works now :+1: I can create my index

About the `make installcheck` command, I did not run it before. I just did now and it _apparently_ works fine.

Although it finishes with the message `All 15 tests passed`, I am a little bit puzzled by few errors visible in the output. But maybe these errors are expected by some tests....just to be sure, can you take a look at the output I got (see below) and tell me whether it is normal or not ?

> /usr/local/lib/postgresql/pgxs/src/makefiles/../../src/test/regress/pg_regress --inputdir=./ --bindir='/usr/local/bin'    --dbname=contrib_regression init tables points euler circle line ellipse poly path box index contains_ops contains_ops_compat bounding_box_gist gnomo
(using postmaster on Unix socket, default port)
============== dropping database "contrib_regression" ==============
DROP DATABASE
NOTICE:  database "contrib_regression" does not exist, skipping
============== creating database "contrib_regression" ==============
CREATE DATABASE
ALTER DATABASE
============== running regression test queries        ==============
test init                     ... ok
test tables                   ... ok
test points                   ... ok
test euler                    ... ok
test circle                   ... ok
test line                     ... ok
test ellipse                  ... ok
test poly                     ... 2018-08-02 10:31:29.494 CEST [88] ERROR:  spherepoly_in: more than two points needed at character 14
2018-08-02 10:31:29.494 CEST [88] STATEMENT:  SELECT spoly '{(10d,0d),(10d,1d)}';
2018-08-02 10:31:29.494 CEST [88] ERROR:  spherepoly_from_array: a line segment overlaps or polygon too large at character 14
2018-08-02 10:31:29.494 CEST [88] STATEMENT:  SELECT spoly '{(0d,0d),(10d,10d),(0d,10d),(10d,0d)}';
ok
test path                     ... ok
test box                      ... ok
test index                    ... ok
test contains_ops             ... ok
test contains_ops_compat      ... ok
test bounding_box_gist        ... ok
test gnomo                    ... ok
>
> ======================
 All 15 tests passed. 
======================
